homeassistant/components/fitbit/sensor.py

```diff
@@ -660,70 +653,83 @@ def __init__(
         api: FitbitApi,
         user_profile_id: str,
         description: FitbitSensorEntityDescription,
-        device: FitbitDevice | None = None,
-        units: str | None = None,
-        enable_default_override: bool = False,
+        units: str | None,
+        enable_default_override: bool,
     ) -> None:
         """Initialize the Fitbit sensor."""
         self.entity_description = description
         self.api = api
-        self.device = device
 
         self._attr_unique_id = f"{user_profile_id}_{description.key}"
-        if device is not None:
-            self._attr_name = f"{device.device_version} Battery"
-            self._attr_unique_id = f"{self._attr_unique_id}_{device.id}"
 
         if units is not None:
             self._attr_native_unit_of_measurement = units
 
         if enable_default_override:
             self._attr_entity_registry_enabled_default = True
 
+    async def async_update(self) -> None:
+        """Get the latest data from the Fitbit API and update the states."""
+        try:
+            result = await self.api.async_get_latest_time_series(
+                self.entity_description.key
+            )
+        except FitbitApiException:
```

Comment:
If there's a `FitbitAuthException` here we should start a reauth flow. We can do that directly or reload the integration to have it try again and start a reauth flow from `async_setup_entry`.

Reply:
Fixed and added test coverage for this case and the same case in the other sensor using the update coordiantor.